File: cpp/Level/6/1157.cpp
#include <iostream>

using namespace std;

int main() {
    string word;
    int index[26] = { 0 };
    int max = 0;
    char c = '?';

    cin >> word;

    for (char ch : word) {
        index[toupper(ch) - 'A']++;
    }

    for (int i = 0; i <= 'Z' - 'A'; i++) {
        if (index[i] > max) {
            max = index[i];
            c = i + 'A';
        } else if (index[i] == max) {
            c = '?';
        }
    }

    cout << c;
    
    return 0;
}

File: python/Level/6/1157.py
word = input().upper()
word_list = list(set(word))
count = []

for i in word_list:
    word_count = word.count(i)
    count.append(word_count)
    
if count.count(max(count)) > 1:
    print("?")
else:
    print(word_list[(count.index(max(count)))])